src/synthesizer_grids/grid_io.py

```diff
@@ -428,9 +433,11 @@ def write_grid_common(
 
         # Write out each axis array
         for axis_key, axis_arr in axes.items():
-            # Handled unitless, logged and linear axes gracefully
-            if "log" in axis_key or not isinstance(axis_arr, unyt_array):
+            # Determine whether the axis should be logged based on log_on_read
+            if log_on_read.get(axis_key, True):
+                axis_arr = np.log10(axis_arr)
```

Comment:
I think there's a bit of a misunderstanding. We want to write linear arrays to the datasets and then have a flag on those datasets that tells us if we need to log it when we read it in on the Synthesizer side.